roles/example-cnf-app/tasks/trex/app.yaml

```diff
@@ -86,7 +86,21 @@
       delay: 5
       until: trex_event.resources|length > 0
 
-    - name: Get TestPassed event from TRex
+    - name: Wait for the TRex app to produce TestPassed or TestFailed event
+      pause:
+        seconds: 30
```

Comment:
By default, the duration of the burst is 120s, so I think you can increase this wait :)

Reply:
Thanks for the info, I think we actually wait for the burst to finish while waiting 300 seconds for the `TestCompleted` event to show up, just one task above.

Here, it is just a minor pause between having the `TestCompleted` event and the `TestFailed`/`TestPassed` event, it should be almost immediate, [there are only some log dumps in-between these two](https://github.com/openshift-kni/example-cnf/blob/eac311b98456edfa5b40a7a8512817962d6b810f/trex-container-app/app/pyfiles/trexstats.py#L111-L138).

Reply:
Your comment made me think that we could write this better, searching for both selectors at once `"reason==TestPassed,reason==TestFailed"` and then using `failed_when` for the check if `TestFailed` is present. I'll wait for the current test to finish, to validate the loop deactivation, and then refine this part.